validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -475,6 +484,47 @@ public void sendAggregateAndProof(final SignedAggregateAndProof aggregateAndProo
             });
   }
 
+  @Override
+  public SafeFuture<Optional<SubmitCommitteeSignaturesResult>> sendSyncCommitteeSignatures(
+      final List<SyncCommitteeSignature> syncCommitteeSignatures) {
+
+    final List<SafeFuture<InternalValidationResult>> addedSignatures =
+        syncCommitteeSignatures.stream()
+            .map(ValidateableSyncCommitteeSignature::fromValidator)
+            .map(syncCommitteeSignaturePool::add)
+            .collect(toList());
+
+    return SafeFuture.collectAll(addedSignatures.stream())
+        .thenApply(this::getSendSyncCommitteesResultFromFutures);
+  }
+
+  private Optional<SubmitCommitteeSignaturesResult> getSendSyncCommitteesResultFromFutures(
+      final List<InternalValidationResult> internalValidationResults) {
+    final List<SubmitCommitteeSignatureError> errorList =
+        internalValidationResults.stream()
+            .flatMap(
+                validationResult ->
+                    this.fromInternalValidationResult(
+                        validationResult, internalValidationResults.indexOf(validationResult))
```

Comment:
We probably don't want to use `.indexOf` to get the index here - we'll wind up repeatedly searching the list so if a significant number of invalid signatures are submitted this will become quite slow as it's O(n^2).  Better to just use an old-skool `for (int index = 0; index < internalValidationResults.size(); index++)` kind of loop so we have the index available.